### PR TITLE
[dynamo] Remove workaround for `functools.wraps` in functorch

### LINKED ISSUE #123365: `@functools.wraps` graph breaks in many cases where we should be able to handle it
### 🐛 Describe the bug

Currently, PyTorch crashes if two functions annotated with `@functools.wraps(func)` are composed together. As an example, when adding support for `torch.func.hessian` in https://github.com/pytorch/pytorch/pull/121410, `wraps` is conditionally used  in `jacrev/jacfwd` if dynamo is not compiling:

* `torch.func.eager_transforms.jacfwd`:
https://github.com/pytorch/pytorch/blob/98e5238ad8b23d62f52549c88ee39f7cbacb46d8/torch/_functorch/eager_transforms.py#L697-L703

* `torch.func.eager_transforms.jacrev`:
https://github.com/pytorch/pytorch/blob/98e5238ad8b23d62f52549c88ee39f7cbacb46d8/torch/_functorch/eager_transforms.py#L1223-L1226

This is because Dynamo treats some of those variables internally as `NestedUserFunctionVariable` and tries to reconstruct it in `SkipFunctionsVariable::call_function`
https://github.com/pytorch/pytorch/blob/98e5238ad8b23d62f52549c88ee39f7cbacb46d8/torch/_dynamo/variables/functions.py#L615-L622

The current approach works but at some point one should revert this and fix the issue, as this same bug appeared in other tests as well: https://github.com/pytorch/pytorch/pull/123357

### Versions

PyTorch main branch

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @zou3519 @ydwu4

### PR DESCRIPTION
**Pull request by StrongerXi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142014
* #142000

This is no longer needed after #142000.

Fixes #123365.

D66838774

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]

File: torch/_functorch/eager_transforms.py [hunk 1]
```diff
@@ -761,12 +761,7 @@ def compute_jacobian_preallocate_and_copy():
             return output_input, aux
         return output_input
 
-    # Dynamo does not support HOP composition if their inner function is
-    # annotated with @functools.wraps(...). We circumvent this issue by applying
-    # wraps only if we're not tracing with dynamo.
-    if not torch._dynamo.is_compiling():
-        wrapper_fn = wraps(func)(wrapper_fn)
-
+    wrapper_fn = wraps(func)(wrapper_fn)
```
[the remark this answers is not in the document]
Comment (StrongerXi) [reply torch/_functorch/eager_transforms.py:764]: Ah right, much cleaner, updated.

**Comment by StrongerXi** [comment]
@StrongerXi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66838610).

**Comment by izaitsevfb** [comment]
@pytorchbot merge -f "unrelated failures"

let's try one more time

**Comment by izaitsevfb** [comment]
@pytorchbot merge -f "unrelated failures"

third time's a charm 
(removed "Differential revision" from the second PR in the stack)

**Pull request closed without merging**